### PR TITLE
Update and expand Build System feature module documentation

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Added lots of documentation about the build system:

- Build configuration
- CLI option
- Editor menu description

#### Tests
`improbadoc serve`. PM for live preview.

#### Documentation
Ye

### CODE REVIEW

File: docs/toc.md [hunk 1]
```diff
@@ -19,6 +19,11 @@
         - [2. Build your workers]({{urlRoot}}/projects/myo/build)
 - <h3>Feature Modules</h3>
     - [Overview]({{urlRoot}}/modules/core-and-feature-module-overview)
+    - Build System
+        - [Overview]({{urlRoot}}/modules/build-system/overview)
+        - [Build configuration]({{urlRoot}}/modules/build-system/build-config)
+        - [Build in the editor]({{urlRoot}}/modules/build-system/editor-menu)
```
Comment (jessicafalk) [inline docs/toc.md]:
```suggestion
        - [Build in the Editor]({{urlRoot}}/modules/build-system/editor-menu)
```

File: docs/modules/core-and-feature-module-overview.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ This provides both a testing framework, which you can use to test any other modu
 
 ### Build system module
 
-The `Improbable.Gdk.BuildSystem` module provides an example implementation of how to build your workers. It creates a menu item in your Unity Editor toolbar (menu: **SpatialOS** > **Build...** ) that you can use to easily build your workers. It also generates a [ScriptableObject (Unity documentation)](https://docs.unity3d.com/ScriptReference/ScriptableObject.html) that contains the build configuration for each worker.  This functionality is only available from your Unity Editor.
+This feature module contains a default implementation and tooling in the Unity Editor for building your GDK for Unity workers. See our [detailed documentation for set up and usage instructions]({{urlRoot}}/modules/build-system/overview).
```
Comment (jessicafalk) [inline docs/modules/core-and-feature-module-overview.md]:
```suggestion
This feature module contains tooling for building your GDK for Unity workers inside the Unity Editor. See our [build system documentation]({{urlRoot}}/modules/build-system/overview) for more details on how to set it up and use it.
```

not sure if we need to say default implementation?

File: docs/modules/build-system/overview.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Build System Feature Module
+
+This feature module contains a default implementation and tooling in the Unity Editor for building your GDK for Unity workers.
```
Comment (jessicafalk) [inline docs/modules/build-system/overview.md]:
```suggestion
This feature module provides tooling for building your GDK for Unity workers inside the Unity Editor.
```

at least I think location needs to be at the end of the sentence

File: docs/modules/build-system/overview.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Build System Feature Module
```
Comment (jessicafalk) [inline docs/modules/build-system/overview.md]: probably shouldn't capitalize it all? same everywhere else in the doc

File: docs/modules/build-system/overview.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+This feature module contains a default implementation and tooling in the Unity Editor for building your GDK for Unity workers.
+
+> **Note:** The Build System Feature Module does not have an integration for building non-Unity workers at this time.
+
+## Installation
+
+### 1. Add the package
```
Comment (jessicafalk) [inline docs/modules/build-system/overview.md]: if there is only one step, should we enumerate it? ;)

File: docs/modules/build-system/editor-menu.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# Editor menu
+
+The Build System Feature Module also provides a set of Unity Editor menu items to build your workers from your Unity Editor. These can be accessed via the **SpatialOS** menu item. 
+
+You can build for a specific target: **Build for local** and **Build for cloud** and then by specific worker type, or build all workers.
+
+For example, to build `UnityClient` for a cloud deployment you would select **SpatialOS** > **Build for cloud** > **UnityClient**.
```
Comment (jessicafalk) [inline docs/modules/build-system/editor-menu.md]:
```suggestion
For example, to build the worker type `UnityClient` for a cloud deployment you would select **SpatialOS** > **Build for cloud** > **UnityClient**.
```

File: docs/modules/build-system/editor-menu.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# Editor menu
+
+The Build System Feature Module also provides a set of Unity Editor menu items to build your workers from your Unity Editor. These can be accessed via the **SpatialOS** menu item. 
+
+You can build for a specific target: **Build for local** and **Build for cloud** and then by specific worker type, or build all workers.
```
Comment (jessicafalk) [inline docs/modules/build-system/editor-menu.md]:
```suggestion
You can build for a specific target: **Build for local** and **Build for cloud** and then either select a specific worker type or build all workers for this target.
```

File: docs/modules/build-system/editor-menu.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+# Editor menu
+
+The Build System Feature Module also provides a set of Unity Editor menu items to build your workers from your Unity Editor. These can be accessed via the **SpatialOS** menu item.
```
Comment (jessicafalk) [inline docs/modules/build-system/editor-menu.md]:
```suggestion
The Build System Feature Module provides a set of Unity Editor menu items to build your workers from your Unity Editor. These can be accessed via the **SpatialOS** menu item. 
```

File: docs/modules/build-system/cli.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
+# Command line interface
+
+The build system also exposes a command line interface (CLI) that you can use to start worker builds. This is useful for building your project on a headless machine or in a continuous integration context.
```
Comment (jessicafalk) [inline docs/modules/build-system/cli.md]:
```suggestion
The build system exposes a command line interface (CLI) that you can use to build your GDK for Unity workers. This is useful for building your project on a headless machine or in a continuous integration context.
```

File: docs/modules/build-system/build-config.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+| | |
+|---|---|
+| **Build Option** | **Description** |
+| Build | Denotes whether to build this target or not. |
+| Required | Denotes whether a build failure while building this target should trigger a build-wide failure. |
+| Development | Denotes whether the build should be a development build with debug symbols. |
+| Server build | Denotes whether the worker is running in headless mode. |
+| Compression | Which compression scheme to use in the build. |
+
+> **Note:** All server-workers **must** have a Linux build target enabled for the cloud target because server-workers are ran on Linux machines in cloud deployments.
```
Comment (jessicafalk) [inline docs/modules/build-system/build-config.md]: link to server-worker definition

File: docs/modules/build-system/build-config.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+### Build and Required options
+
+These two options describe the behaviour of a worker-type, deployment target, and platform combination.
```
Comment (jessicafalk) [inline docs/modules/build-system/build-config.md]: worker-type or worker type? I believe we have been mainly using worker type

File: docs/modules/build-system/overview.md (hunk 5 above)
Comment (jamiebrynes7) [reply docs/modules/build-system/overview.md]: Mostly just consistency across the feature modules but can just remove it

File: docs/modules/build-system/build-config.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
+# Build configuration
+
+The Build System feature module uses a scriptable object for its build configuration.
```
Comment (jessicafalk) [inline docs/modules/build-system/build-config.md]: should feature module be capitalized here? ;)

File: docs/modules/build-system/editor-menu.md (hunk 8 above)
Comment (jessicafalk) [reply docs/modules/build-system/editor-menu.md]: marked as resolved, but not changed :P

Comment (jamiebrynes7) [reply docs/modules/build-system/editor-menu.md]: I thought I did? 🤔

File: docs/modules/build-system/build-config.md (hunk 12 above)
Comment (jamiebrynes7) [reply docs/modules/build-system/build-config.md]: To @ElleEss's sadness, yes

**Pull request merged**